homeassistant/components/palazzetti/config_flow.py

```diff
@@ -48,3 +53,46 @@ async def async_step_user(
             data_schema=vol.Schema({vol.Required(CONF_HOST): str}),
             errors=errors,
         )
+
+    async def async_step_dhcp(
+        self, discovery_info: dhcp.DhcpServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle DHCP discovery."""
+
+        LOGGER.debug(
+            "DHCP discovery detected Palazzetti: %s", discovery_info.macaddress
+        )
+
+        self._async_abort_entries_match({CONF_HOST: discovery_info.ip})
+
+        await self.async_set_unique_id(dr.format_mac(discovery_info.macaddress))
+        self._abort_if_unique_id_configured()
+        self._discovered_device = PalazzettiClient(hostname=discovery_info.ip)
+        try:
+            await self._discovered_device.connect()
+        except CommunicationError:
+            LOGGER.exception("Communication error")
+            return self.async_abort(reason="cannot_connect")
```

Comment:
I would move that to confirm, so the user actively gets the error, but don't know what best practice is

Reply:
I have thought of that but:
1. This is how @joostlek does it in the AirGradient integration
2. Even if we put it in `confirm`, when it throws we will abort before we return `async_show_form(...)` and the user experience will be exactly the same

Reply:
Apologies, my point 2 is invalid: we could generate a very generic form without the data coming from the API (name, host), and have it fail when the user clicks "submit". I'm not sure that it is a much better experience, though. If the problem is permanent, why nag the user with notifications to add a device which always fail? And if it is temporary, HA will show the notification later when it works.